SignalRClient/HttpConnection.swift

```diff
@@ -148,6 +148,13 @@ public class HttpConnection: Connection {
     }
 
     fileprivate func transportDidOpen() {
+        // connection is being stopped even though start has not finished yet
+        if (self.state != State.connecting) {
+            self.startDispatchGroup.leave()
+            self.failOpenWithError(error: SignalRError.connectionIsBeingClosed, changeState: false)
+            return
+        }
```

Comment:
I have doubts about this change. I believe the condition to execute this code is not that `self.state != State.connecting` but that `self.state == State.stopped`. I need to think about this more but looking at this now I think the way to handle this is to conditionally change state and only act if this failed.

I have started refactoring the code so that it is possible to test these kinds of scenarios/races.